scripts/check_delta.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2020 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Get different tests delta between two xml files,
+for example, two xml files of same board with different commits,
+or two xml files of same commit on different boards
+"""
+
+import argparse
+import os
+import sys
+import xml.etree.ElementTree as ET
+
+CWD = os.getcwd()
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+            description = "find different tests delta")
+    parser.add_argument("-x", "--xml-file", default = None,
+            action = "append", help = "xml file path. This option "
+            "may be used multiple times.")
+    parser.add_argument("-o", "--output-dir", default = CWD,
+            help = "Output dir")
+
+    return parser.parse_args()
+
+def parse_xml(xmlfiles, output):
+    # get all testcases name
+    tree_1 = ET.parse(xmlfiles[0])
+    testsuite_1 = tree_1.getroot()[0]
+    testcases_1 = testsuite_1.findall("testcase")
+    testcase_names_1 = set()
+    for item in testcases_1:
+        testcase_names_1.add(item.attrib["name"])
+
+    tree_2 = ET.parse(xmlfiles[1])
+    testsuite_2 = tree_2.getroot()[0]
+    testcases_2 = testsuite_2.findall("testcase")
+    testcase_names_2 = set()
+    for item in testcases_2:
+        testcase_names_2.add(item.attrib["name"])
+
+    # get all common testcases name
+    common_test_names = testcase_names_1.intersection(testcase_names_2)
+
+    # remove common testcases
+    for item in testcases_1:
+        if item.attrib["name"] in common_test_names:
+            testsuite_1.remove(item)
+            testsuite_1.attrib["tests"] = str(int(testsuite_1.attrib["tests"]) - 1)
+            if item.find("skipped") is not None and item.find("skipped").attrib["type"] == "skipped":
+                testsuite_1.attrib["skip"] = str(int(testsuite_1.attrib["skip"]) - 1)
+            elif item.find("errors") is not None and item.find("errors").attrib["type"] == "errors":
+                testsuite_1.attrib["errors"] = str(int(testsuite_1.attrib["errors"]) - 1)
+            elif item.find("failures") is not None and item.find("failures").attrib["type"] == "failures":
+                testsuite_1.attrib["failures"] = str(int(testsuite_1.attrib["failures"]) - 1)
+            else:
+                pass
+
```

Comment:
Please see the link below. It seems to me that there is easier way to find difference between the two lists.

https://www.geeksforgeeks.org/python-difference-two-lists/

Reply:
https://github.com/Shoobx/xmldiff or this tool